oximeter/db/src/lib.rs

```diff
@@ -70,6 +77,197 @@ pub enum Error {
 
     #[error("The field comparison {op} is not valid for the type {ty}")]
     InvalidFieldCmp { op: String, ty: FieldType },
+
+    #[error("Invalid timeseries name")]
+    InvalidTimeseriesName,
+}
+
+/// A timeseries name.
+///
+/// Timeseries are named by concatenating the names of their target and metric, joined with a
+/// colon.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    JsonSchema,
+)]
+pub struct TimeseriesName(String);
```

Comment:
We may want to use `serde(try_from = "&str")` here to use the validation methods when deserializing. It's probably also useful to encode the constraints on these names in a custom `JsonSchema` implementation, so that some amount of client-side validation can be done.

Reply:
Done in 9919858.